src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs

```diff
@@ -325,7 +331,7 @@ internal ValueTask FinishHandshakeAsync(QuicServerConnectionOptions options, str
             _sslConnectionOptions = new SslConnectionOptions(
                 this,
                 isClient: false,
-                targetHost: null,
+                targetHost: targetHost,
```

Comment:
Can we receive punycoded SNI here? If so, should we decode it?
Looks like `SslStream` doesn't do anything with the received value, but I don't know whether that's an oversight, or the layer underneath does it, or it just cannot happen.

Reply:
Looks like SslStream already receives decoded value (there are tests in SslStreamSniTests that test this path). I will add tests for QUIC as well

Reply:
aand MsQuic is already handling punycode for us, so we can skip this part